amethyst_controls/src/systems.rs

```diff
@@ -80,6 +80,32 @@ where
     }
 }
 
+/// The system that manages the arc ball movement;
+pub struct ArcBallMovementSystem;
+
+impl<'a> System<'a> for ArcBallMovementSystem {
+    type SystemData = (
+        WriteStorage<'a, Transform>,
+        ReadStorage<'a, ArcBallCameraTag>,
+    );
+
+    fn run(&mut self, (mut transforms, tags): Self::SystemData) {
+        let mut position = None;
```

Comment:
There's a slightly better solution that also resolves my concern below: you can use `RestrictedStorage` to iterate over the transforms and then set the position in the same iteration of the loop. I'll come back to you and explain in more detail later.

Reply:
I though that it would be only one arcball camera (I got no reference about a game having several arcball camera other that split screen game). The exemple is planned, but at 1h30 am, it was difficult to explain to my partner I was still working on code ^^'
The restricted storage could be something.